### PR TITLE
Added tests for submissions.models.Entry:get_entry_feature(). some fail.

### PR DESCRIPTION
**Pull request by toolness**
I wasn't sure about the behavior of `get_entry_feature()` so I wrote some tests to validate some assumptions. Currently the results are:

```
.F..F
======================================================================
FAIL: test_is_none_when_video_url_is_not_allowed_2 (gameon.submissions.test_models.EntryFeatureVideoUrlTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/atul/Documents/mozilla/gameon/gameon/submissions/test_models.py", line 31, in test_is_none_when_video_url_is_not_allowed_2
    self.assertIsEmpty(video_url="http://mysite.org/youtube")
  File "/Users/atul/Documents/mozilla/gameon/gameon/submissions/test_models.py", line 15, in assertIsEmpty
    (self._repr(m), repr(html)))
AssertionError: Expected Entry(video_url='http://mysite.org/youtube').get_entry_feature() to return None, but got '<a href="https://mysite.org/youtube">https://mysite.org/youtube</a>'

======================================================================
FAIL: test_is_oembed_when_video_url_is_youtube_shorturl (gameon.submissions.test_models.EntryFeatureVideoUrlTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/atul/Documents/mozilla/gameon/gameon/submissions/test_models.py", line 34, in test_is_oembed_when_video_url_is_youtube_shorturl
    self.assertIsEmbed(video_url="http://youtu.be/ITTxTCz4Ums")
  File "/Users/atul/Documents/mozilla/gameon/gameon/submissions/test_models.py", line 23, in assertIsEmbed
    self.fail(prefix + "None")
AssertionError: Expected Entry(video_url='http://youtu.be/ITTxTCz4Ums').get_entry_feature() to return an <iframe> tag, but got None

----------------------------------------------------------------------
Ran 5 tests in 2.123s

FAILED (failures=2)
```

So, it looks like links with a `youtu.be` domain (which are currently the default when clicking "share" for youtube videos) are rejected, and it looks like URL with the words `youtube` or `vimeo` _anywhere_ in them are linkified (not embedded)--this isn't horrible, but seems a bit unexpected and I suspect the preferred behavior would be to fall-back to the thumbnail URL instead of showing the hyperlink.

One way to resolve this is to use [urlparse.parse()](http://docs.python.org/2/library/urlparse.html#urlparse.urlparse) to excise the domain and then compare it against a whitelist, rather than looking for the whitelisted words anywhere in the URL. This personally makes my internal "spider-sense" go off way less when thinking about potential security vulnerabilities, and also results in more predictable behavior... but it's just a suggestion.

### CODE REVIEW

**Comment by rossbruniges** [comment]
Very cool - thanks for that.

Looking at the tests briefly I think we've actually got the expected behaviour. For video_url we're expecting a link to an external video - so a user can click it and see it. We're not asking for or expecting an embeded player so would treat that as invalid and not try to create a player for it....

**Comment by rossbruniges** [comment]
Or maybe not - I think we can make the whitelist a bit more robust and sort these out.

File: .gitmodules [hunk 1]
```diff
@@ -1,6 +1,3 @@
 [submodule "vendor"]
 	path = vendor
 	url = git://github.com/mozilla/playdoh-lib.git
-[submodule "vendor-local/src/micawber"]
```
Comment (rossbruniges) [inline .gitmodules:4]: Thanks for cleaning up :)

File: gameon/submissions/models.py [hunk 2]
```diff
@@ -13,6 +13,41 @@
 from gameon.base.utils import _upload_path
 from managers import ChallengeManager
 
+EMBED_DEFAULTS = {
```
Comment (rossbruniges) [inline gameon/submissions/models.py]: I'm not sure which is best but I kinda like these things in the settings.py files

File: gameon/submissions/models.py [hunk 3]
```diff
@@ -13,6 +13,41 @@
...
+    # http://apiblog.youtube.com/2010/07/new-way-to-embed-youtube-videos.html
+    'youtube': {
+        'regexps': [
+            re.compile(r'^https?://youtu\.be/(?P<VIDEO_ID>\w+)$'),
+            re.compile(r'^https?://www.youtube.com/watch\?v=(?P<VIDEO_ID>\w+)$')
+        ],
+        'embed': """<iframe class="youtube-player" type="text/html" width="%(WIDTH)d" height="%(HEIGHT)d" src="https://www.youtube.com/embed/%(VIDEO_ID)s" frameborder="0"></iframe>"""
+    },
+    # Embed code obtained through inference.
+    'vimeo': {
```
Comment (rossbruniges) [inline gameon/submissions/models.py:29]: Is this based on anything official like the youtube stuff above?

File: gameon/submissions/models.py [hunk 4]
```diff
@@ -13,6 +13,41 @@
 from gameon.base.utils import _upload_path
 from managers import ChallengeManager
 
+EMBED_DEFAULTS = {
+    'WIDTH': 640,
+    'HEIGHT': 385
+}
+
+URL_TO_EMBED_MAPPINGS = {
```
Comment (rossbruniges) [inline gameon/submissions/models.py:16]: This makes me feel like the don't need the settings.ALLOWED_OMEMBED_SITES? Or maybe the values in that could be tweaked a little?

Comment (toolness) [inline gameon/submissions/models.py:16]: Yeah, I can remove that. Was thinking about doing that when writing the code, but felt lazy... :)

File: gameon/submissions/models.py (hunk 3 above)
Comment (toolness) [inline gameon/submissions/models.py:29]: Unfortunately not--totally reverse-engineered :(

Comment (toolness) [inline gameon/submissions/models.py:29]: But I figure if it fails ... at worst it's just a blank iframe, and not some kind of XSS attack?

File: gameon/submissions/models.py (hunk 2 above)
Comment (toolness) [inline gameon/submissions/models.py]: Yeah, I can do that. Was thinking about doing that when writing the code, but felt lazy... :)

**Pull request merged**